bpo-45132 Remove deprecated __getitem__ methods

hugovk opened the pull request:
Remove deprecated `__getitem__` methods of `xml.dom.pulldom.DOMEventStream`, `wsgiref.util.FileWrapper` and `fileinput.FileInput`, deprecated since Python 3.9.

<!--
Thanks for your contribution!
Please read this comment in its entirety. It's quite important.

# Pull Request title

It should be in the following format:

```
bpo-NNNN: Summary of the changes made
```

Where: bpo-NNNN refers to the issue number in the https://bugs.python.org.

Most PRs will require an issue number. Trivial changes, like fixing a typo, do not need an issue.

# Backport Pull Request title

If this is a backport PR (PR made against branches other than `main`),
please ensure that the PR title is in the following format:

```
[X.Y] <title from the original PR> (GH-NNNN)
```

Where: [X.Y] is the branch name, e.g. [3.6].

GH-NNNN refers to the PR number from `main`.

-->


<!-- issue-number: [bpo-45132](https://bugs.python.org/issue45132) -->
https://bugs.python.org/issue45132
<!-- /issue-number -->

hugovk committed:
bpo-45132 Remove deprecated __getitem__ methods

Remove deprecated __getitem__ methods of xml.dom.pulldom.DOMEventStream,
wsgiref.util.FileWrapper and fileinput.FileInput, deprecated since Python 3.9.

serhiy-storchaka commented on Doc/library/fileinput.rst:
```diff
@@ -171,8 +170,8 @@ available for subclassing as well:
 
-   .. deprecated:: 3.8
-      Support for :meth:`__getitem__` method is deprecated.
+   .. deprecated-removed:: 3.8 3.11
```
```suggestion
   .. versionchanged:: 3.11
```
and move it below `versionchanged:: 3.8`.

`deprecated-removed` is only used for future planned removing. After removing it is just `versionchanged`.

serhiy-storchaka commented on Doc/library/fileinput.rst:
```diff
@@ -147,8 +147,7 @@ available for subclassing as well:
    functions of the same name in the module. In addition it has a
-   :meth:`~io.TextIOBase.readline` method which returns the next input line,
-   and a :meth:`__getitem__` method which implements the sequence behavior.
+   :meth:`~io.TextIOBase.readline` method which returns the next input line.
```
"```In addition it is :term:`iterable` and has a``` ..."

The only purpose of `__getitem__` was supporting iteration (before introducing the iterable protocol).

serhiy-storchaka commented on Doc/library/wsgiref.rst:
```diff
@@ -151,8 +151,7 @@ also provides these miscellaneous utilities:
    A wrapper to convert a file-like object to an :term:`iterator`.  The resulting objects
-   support both :meth:`__getitem__` and :meth:`__iter__` iteration styles, for
-   compatibility with Python 2.1 and Jython. As the object is iterated over, the
+   support the :meth:`__iter__` iteration style. As the object is iterated over, the
```
```suggestion
   are :term`iterable`\ s. As the object is iterated over, the
```

serhiy-storchaka commented on Doc/library/wsgiref.rst:
```diff
@@ -173,8 +172,8 @@ also provides these miscellaneous utilities:
-   .. deprecated:: 3.8
-      Support for :meth:`sequence protocol <__getitem__>` is deprecated.
+   .. deprecated-removed:: 3.8 3.11
+      Support for sequence protocol :meth:`__getitem__` has been removed.
```
```suggestion
      Support for :meth:`__getitem__` method has been removed.
```

serhiy-storchaka commented on Doc/library/wsgiref.rst:
```diff
@@ -173,8 +172,8 @@ also provides these miscellaneous utilities:
 
-   .. deprecated:: 3.8
-      Support for :meth:`sequence protocol <__getitem__>` is deprecated.
+   .. deprecated-removed:: 3.8 3.11
```
```suggestion
   .. versionchanged:: 3.11
```

serhiy-storchaka commented on Doc/library/xml.dom.pulldom.rst:
```diff
@@ -114,8 +114,8 @@ DOMEventStream Objects
-   .. deprecated:: 3.8
-      Support for :meth:`sequence protocol <__getitem__>` is deprecated.
+   .. deprecated-removed:: 3.8 3.11
+      Support for sequence protocol :meth:`__getitem__` has been removed.
```
```suggestion
   .. versionchanged:: 3.11
      Support for :meth:`__getitem__` method has been removed.
```

serhiy-storchaka reviewed (commented):
LGTM, but I have some documentation nitpicks.

hugovk committed:
bpo-45132 Update documentation

`deprecated-removed` is only used for future planned removing. After removing it is just `versionchanged`.

Co-authored-by: Serhiy Storchaka <storchaka@gmail.com>

hugovk committed:
bpo-45132 Update docs

Insert "In addition it is :term:`iterable` and has a ..."

deprecated-removed is only used for future planned removing.
After removing it is just versionchanged.

serhiy-storchaka reviewed (commented):
After removing deprecated features we remove `deprecated` directives.

serhiy-storchaka commented on Doc/library/fileinput.rst:
```diff
@@ -183,6 +182,9 @@ available for subclassing as well:
    .. versionchanged:: 3.11
       The ``'rU'`` and ``'U'`` modes have been removed.
 
+   .. versionchanged:: 3.11
```
BTW, two directives for different changes in the same version can be merged.

hugovk committed:
bpo-45132 Docs: Remove old deprecated directives, we have new versionchanged directives

serhiy-storchaka reviewed (approved):
LGTM
